dlt/common/configuration/specs/aws_credentials.py

```diff
@@ -183,7 +183,8 @@ def _from_session(self, session: Any) -> Any:
             return None
         self.aws_access_key_id = default.access_key
         self.aws_secret_access_key = TSecretStrValue(default.secret_key)
-        self.aws_session_token = TSecretStrValue(default.token)
+        if default.token is not None:
+            self.aws_session_token = TSecretStrValue(default.token)
```

Comment:
Let's change this to:

`self.aws_session_token = cast(TSecretStrValue, default.token)`

You can do the same for the line above!